### PR TITLE
Replace OverlapReal with ShortReal.

### LINKED ISSUE #1447: Replace `OverlapReal` with `ShortReal`.
### Description

Global find and replace `OverlapReal` with `ShortReal`.

### Motivation and context

#1444 makes the mixed precision types available for the entire codebase, where `OverlapReal` is specific to HPMC. 

Note: To avoid merge conflicts, defer this task until just before the v4 release and preferably when there is no ongoing work on HPMC based on `trunk-minor`.

### PR DESCRIPTION
**Pull request by joaander**
## Description

* Replace all instances of `OverlapReal` with `ShortReal`.
* Remove the now unused `HPMCPrecisionSetup.h`.

## Motivation and context

`ShortReal` is the new mechanism for managing mixed precision builds. It replaces the old `OverlapReal` type which was limited to HPMC.

Resolves #1447.

## How has this been tested?

Local builds pass.

## Change log

```
Removed:

* ``hpmc::OverlapReal`` C++ data type. Use ``ShortReal``.
```

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/CONTRIBUTING.rst).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

File: hoomd/hpmc/ShapeConvexPolygon.h
Comment (b-butler) [inline hoomd/hpmc/ShapeConvexPolygon.h]: Is there a reason we keep this?

File: hoomd/hpmc/ShapeSpheropolyhedron.h
Comment (b-butler) [inline hoomd/hpmc/ShapeSpheropolyhedron.h]: Same comment as before.

File: hoomd/hpmc/test/test_convex_polygon.cc
Comment (b-butler) [inline hoomd/hpmc/test/test_convex_polygon.cc]: Why this comment?

File: hoomd/hpmc/test/test_convex_polyhedron.cc
Comment (b-butler) [inline hoomd/hpmc/test/test_convex_polyhedron.cc]: More commented code.

File: hoomd/hpmc/test/test_simple_polygon.cc
Comment (b-butler) [inline hoomd/hpmc/test/test_simple_polygon.cc]: Not even used commented function.

File: hoomd/hpmc/test/test_spheropolyhedron.cc
Comment (b-butler) [inline hoomd/hpmc/test/test_spheropolyhedron.cc]: More commented code.

File: hoomd/hpmc/ShapeConvexPolyhedron.h
Comment (b-butler) [inline hoomd/hpmc/ShapeConvexPolyhedron.h]: Commented out code.

File: hoomd/hpmc/ShapeSphinx.h
Comment (b-butler) [inline hoomd/hpmc/ShapeSphinx.h:1102]: More commented code.

File: hoomd/hpmc/SphinxOverlap.h
Comment (b-butler) [inline hoomd/hpmc/SphinxOverlap.h:3337]: Commented code.

**Review by b-butler — APPROVED** [review]
Looks good a simple find and replace for the most part. I left comments where I found changes in commented code; there was quite a bit. I don't want to try and hold up this PR for that, so if you feel like removing it before merging you can, but otherwise :+1: .

**Review by cbkerr — APPROVED** [review]
:)

**Comment by joaander** [comment]
The commented code fell into two categories: 1) Alternate implementations of methods that ended up being slower when benchmarked or 2) Methods that wrote out visual information from unit tests so that the shape overlap checks could be inspected visually. I removed these.

**Pull request merged**
